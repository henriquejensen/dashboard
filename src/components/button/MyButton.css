.my-button-circle {
    padding: 2px 7px 0px 7px;
    background: #F0AD4E;
    color: white;
    border-color: #F0AD4E;
    border-radius: 15px;
}

.my-button-circle:hover {
    color: white;
    background: #f9cd8f;
    border-color: #f9cd8f;
}

.my-button-circle:focus {
    color: white;
    background: #f9cd8f;
    border-color: #f9cd8f;
}