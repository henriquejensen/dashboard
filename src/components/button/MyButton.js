import React from 'react';
import { Button } from "react-bootstrap";

const MyButton = (props) => {
    let tooltip = props.tooltip ? props.tooltip : "";
    let labelButton = props.label ? props.label : "";
    let params = props.params;
    let myButtonClass = props.myButtonClass ? props.myButtonClass : "mapa-button";
    let myButtonStyle = props.myButtonStyle ? props.myButtonStyle : "info";
    let myButtonText = props.myButtonText ? props.myButtonText : <i className='fa fa-search'/>;

    return (
        <span>
            <a data-tip data-for={tooltip}>
                <Button bsStyle={myButtonStyle} className={myButtonClass} onClick={params ? () => props.onClickButton(...params) : props.onClickButton}>
                    {myButtonText}
                </Button>
            </a>
            {labelButton}
        </span>
    );
};

export default MyButton;