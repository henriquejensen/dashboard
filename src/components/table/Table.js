import React, { Component } from "react";

export default class Table extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div>
                {this.props.title ?
                    <h4>{this.props.title}</h4>
                : ""}
                <table className="table table-striped table-hover table-responsive">
                    <thead>
                        <tr>
                            {this.props.fields ? 
                                this.props.fields.map((field, index) => {
                                    return <th key={index}>{field}</th>
                                })
                            : ""}
                        </tr>
                    </thead>
                    
                    {this.props.children}
                    
                </table>
            </div>
        )
    }

}
