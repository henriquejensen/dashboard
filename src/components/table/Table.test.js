import React from 'react';
import renderer from "react-test-renderer";
import Table from "./MyTable";
import MyButton from "../button/MyButton";

describe("<MyTable/>", () => {
    let title="Teste Title";
    let columns=[
        {id:"Column_1", name:"Label_1"},
        {id:"Column_2", name:"Label_2"},
        {id:"Column_3", name:"Label_3"},
        {id:"Column_4", name:"Label_4"},
    ];
    let elements = [
        {Column_1:"Teste_1", Column_2:"Teste_2", Column_3:"Teste_3", Column_4:"Teste_4"},
        {Column_1:"Teste_5", Column_2:"Teste_6", Column_3:"Teste_7", Column_4:"Teste_8"},
        {Column_1:"Teste_9", Column_2:"Teste_10", Column_3:"Teste_12", Column_4:"Teste_13"}
    ];
    it('gerando o elemento sem quebrar', () => {
        const tree = renderer.create(<Table title={title} fields={columns} elements={elements} />);

        expect(tree.toJSON()).toMatchSnapshot();
    });
    
    /*columns.push({key:"Column_5", label:"Label_5"});
    elements = [
        {Label_1:"Teste_1", Label_2:"Teste_2", Label_3:"Teste_3", Label_4:"Teste_4", btn:<MyButton/>},
        {Label_1:"Teste_5", Label_2:"Teste_6", Label_3:"Teste_7", Label_4:"Teste_8", btn:<MyButton/>},
        {Label_1:"Teste_9", Label_2:"Teste_10", Label_3:"Teste_12", Label_4:"Teste_13", btn:<MyButton/>}
    ];
    it('renderizando um botao na linha da tabela', () => {
        const tree = shallow(<Table title={title} fields={columns} elements={elements} />);

        console.log("TTT", tree.find("button"))
        expect(tree.find("button").length).toBe(3);
    });*/
});