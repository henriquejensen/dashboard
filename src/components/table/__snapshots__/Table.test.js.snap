// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyTable/> gerando o elemento sem quebrar 1`] = `
<div>
  <h4>
    Teste Title
  </h4>
  <div
    className="table-responsive"
  >
    <table
      className="my-table table table-striped table-hover"
    >
      <thead>
        <th>
          <th>
            Label_1
          </th>
          <th>
            Label_2
          </th>
          <th>
            Label_3
          </th>
          <th>
            Label_4
          </th>
        </th>
      </thead>
      <tbody>
        <tr>
          <td>
            Teste_1
          </td>
          <td>
            Teste_2
          </td>
          <td>
            Teste_3
          </td>
          <td>
            Teste_4
          </td>
        </tr>
        <tr>
          <td>
            Teste_5
          </td>
          <td>
            Teste_6
          </td>
          <td>
            Teste_7
          </td>
          <td>
            Teste_8
          </td>
        </tr>
        <tr>
          <td>
            Teste_9
          </td>
          <td>
            Teste_10
          </td>
          <td>
            Teste_12
          </td>
          <td>
            Teste_13
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
